piffle/iiif.py

```diff
@@ -108,3 +108,125 @@ def format(self, image_format):
         img = self.get_copy()
         img.image_options['format'] = image_format
         return img
+
+    def parse_from_url(self, url):
+        '''Parses Image API parameters from URL provided
+        Per http://iiif.io/api/image/2.0/#image-request-uri-syntax, using slashes to navigate URL'''
+        
+        url_components = url.split('/')
+        
+        _quality, _format = url_components[-1].split('.')
+        _rotation = url_components[-2]
+        _size = url_components[-3]
+        _region = url_components[-4]
+        _image_id = url_components[-5]
+        _api_endpoint = '/'.join(url_components[:-6])
+        
+        # reinit
+        self.__init__(api_endpoint=_api_endpoint, image_id=_image_id, region=_region,
+                     size=_size, rotation=_rotation, quality=_quality, format=_format)
+                     
+    def api_params_as_dict(self):
+        
+        '''
+        Returns dictionary of region, size, and rotation parameter strings
+        parsed as dictionaries.
+        '''
+
+        return {
+            'region':self._region_as_dict(),
+            'size':self._size_as_dict(),
+            'rotation':self._rotation_as_dict()            
+        }
+
+    # methods to derive python dictionaries from IIIF strings
+    def _region_as_dict(self):
```

Comment:
Any reason the sub dict methods shouldn't be exposed publicly? Seems like they could be useful.